### PR TITLE
Remove pandas dependency [Tech debt]

### PR DESCRIPTION
**Pull request by vembar**
We use pandas to simply print the inspect table. This PR replaces the pandas dataframe with a regular list and a print function to print the table. 

```
>>> nlp.inspect('turn off the lights', domain='smart_home')
Inspecting domain classification
Feature                                          	Value 	Pred_W(smart_home)	Pred_P   	Gold_W(smart_home)	Gold_P   	Diff
bag_of_words|edge:left|length:1|ngram:turn       	1     	[1.0427]          	[1.0427] 	[1.0427]          	[1.0427] 	[0.]
bag_of_words|edge:left|length:2|ngram:turn off   	1     	[-1.0536]         	[-1.0536]	[-1.0536]         	[-1.0536]	[0.]
bag_of_words|edge:right|length:1|ngram:lights    	1     	[0.6844]          	[0.6844] 	[0.6844]          	[0.6844] 	[0.]
bag_of_words|edge:right|length:2|ngram:the lights	1     	[0.3232]          	[0.3232] 	[0.3232]          	[0.3232] 	[0.]
bag_of_words|length:1|ngram:lights               	1     	[5.0197]          	[5.0197] 	[5.0197]          	[5.0197] 	[0.]
bag_of_words|length:1|ngram:off                  	1     	[2.2332]          	[2.2332] 	[2.2332]          	[2.2332] 	[0.]
bag_of_words|length:1|ngram:the                  	1     	[1.37]            	[1.37]   	[1.37]            	[1.37]   	[0.]
bag_of_words|length:1|ngram:turn                 	1     	[2.0373]          	[2.0373] 	[2.0373]          	[2.0373] 	[0.]
bag_of_words|length:2|ngram:off the              	1     	[-0.0626]         	[-0.0626]	[-0.0626]         	[-0.0626]	[0.]
bag_of_words|length:2|ngram:the lights           	1     	[1.2159]          	[1.2159] 	[1.2159]          	[1.2159] 	[0.]
bag_of_words|length:2|ngram:turn off             	1     	[-1.0626]         	[-1.0626]	[-1.0626]         	[-1.0626]	[0.]
exact|query:<turn off the lights>                	10    	[-0.0091]         	[-0.0905]	[-0.0091]         	[-0.0905]	[0.]
in_gaz|type:city|gaz_freq_bin:0                  	0.25  	[-0.7569]         	[-0.1892]	[-0.7569]         	[-0.1892]	[0.]
in_gaz|type:city|gaz_freq_bin:1                  	0.25  	[1.1835]          	[0.2959] 	[1.1835]          	[0.2959] 	[0.]
in_gaz|type:city|gaz_freq_bin:4                  	0.25  	[0.8377]          	[0.2094] 	[0.8377]          	[0.2094] 	[0.]
in_vocab:IV|freq_bin:0                           	0.25  	[0.5286]          	[0.1322] 	[0.5286]          	[0.1322] 	[0.]
in_vocab:IV|freq_bin:2                           	0.25  	[-1.2738]         	[-0.3184]	[-1.2738]         	[-0.3184]	[0.]
in_vocab:IV|freq_bin:3                           	0.3962	[0.7585]          	[0.3005] 	[0.7585]          	[0.3005] 	[0.]
in_vocab:IV|in_gaz|type:city|gaz_freq_bin:0      	0.25  	[-0.7586]         	[-0.1896]	[-0.7586]         	[-0.1896]	[0.]
in_vocab:IV|in_gaz|type:city|gaz_freq_bin:1      	0.25  	[3.1868]          	[0.7967] 	[3.1868]          	[0.7967] 	[0.]
in_vocab:IV|in_gaz|type:city|gaz_freq_bin:4      	0.25  	[0.8377]          	[0.2094] 	[0.8377]          	[0.2094] 	[0.]

```

```
>>> nlp.inspect('hello!', domain='greeting', intent='exit')
Inspecting domain classification
Feature                          	Value	Pred_W(greeting)	Pred_P  	Gold_W(greeting)	Gold_P  	Diff
bag_of_words|length:1|ngram:hello	1    	[0.4763]        	[0.4763]	[0.4763]        	[0.4763]	[0.]
exact|query:<hello>              	10   	[0.5885]        	[5.8847]	[0.5885]        	[5.8847]	[0.]
in_vocab:IV|freq_bin:5           	1.0  	[4.1858]        	[4.1858]	[4.1858]        	[4.1858]	[0.]

Inspecting intent classification
Feature                          	Value	Pred_W(greet)	Pred_P	Gold_W(exit)	Gold_P  	Diff    
bag_of_words|length:1|ngram:hello	1    	[0.]         	[0.]  	[0.3218]    	[0.3218]	[0.3218]
exact|query:<hello>              	10   	[0.]         	[0.]  	[0.5181]    	[5.181] 	[5.181] 
in_vocab:IV|freq_bin:1           	1.0  	[0.]         	[0.]  	[0.5088]    	[0.5088]	[0.5088]

```

### CODE REVIEW

**Comment by serapio** [comment]
Nice.
1. There are a bunch of references to DataFrame in the docstrings.
2. Does anyone remember why the intercepts don't show up in this breakdown? I'm pretty sure it was showing in the WB2 version of this, but doesn't here.

**Comment by minhtuev** [comment]
You will need to apply black and fix lint:
```
would reformat /home/runner/work/mindmeld/mindmeld/mindmeld/components/nlp.py
would reformat /home/runner/work/mindmeld/mindmeld/mindmeld/models/text_models.py
Oh no! 💥 💔 💥
2 files would be reformatted, 110 files would be left unchanged.
mindmeld/models/text_models.py:318:29: E126 continuation line over-indented for hanging indent
mindmeld/models/text_models.py:323:25: E126 continuation line over-indented for hanging indent
mindmeld/models/text_models.py:331:29: E126 continuation line over-indented for hanging indent
mindmeld/models/text_models.py:338:25: E126 continuation line over-indented for hanging indent
************* Module mindmeld.components.nlp
mindmeld/components/nlp.py:643:4: R1711: Useless return at end of function or method (useless-return)
```

File: mindmeld/components/nlp.py [hunk 1]
```diff
@@ -639,6 +639,18 @@ def extract_allowed_intents(self, allowed_intents):
 
         return nlp_components
 
+    @staticmethod
+    def print_inspect_stats(stats):
+        """
+        Prints formatted output matrix
```
Comment (minhtuev) [inline mindmeld/components/nlp.py:645]: If this is a public method we should provide some documentation on the stats argument (what is its format, etc)

File: mindmeld/models/text_models.py [hunk 2]
```diff
@@ -315,45 +314,35 @@ def inspect(self, example, gold_label=None, dynamic_resource=None):
...
+                        weight.round(4),
+                        product.round(4),
+                        '-', '-', '-'
+                ]
+
             else:
                 gold_w = self._get_feature_weight(feat_name, gold_class)
                 gold_p = feat_value * gold_w
                 diff = gold_p - product
                 # pylint: disable=no-member
```
Comment (minhtuev) [inline mindmeld/models/text_models.py:329]: you can remove this pylint now

File: mindmeld/models/text_models.py [hunk 3]
```diff
@@ -315,45 +314,35 @@ def inspect(self, example, gold_label=None, dynamic_resource=None):
 
             if gold_class is None:
                 # pylint: disable=no-member
```
Comment (minhtuev) [inline mindmeld/models/text_models.py:316]: remove pylint

File: mindmeld/components/nlp.py (hunk 1 above)
Comment (minhtuev) [reply mindmeld/components/nlp.py:645]: if stats is an empty [], what will the output to str be?

Comment (vembar) [reply mindmeld/components/nlp.py:645]: It would just print two new lines.

File: mindmeld/components/nlp.py [hunk 4]
```diff
@@ -661,17 +673,15 @@ def inspect(self, markup, domain=None, intent=None, dynamic_resource=None):
             domain_inspection = self.domain_classifier.inspect(
                 query, domain=domain, dynamic_resource=dynamic_resource
             )
```
Comment (serapio) [inline mindmeld/components/nlp.py:675]: The doc strings for domain_classifier.inspect(), intent_classifier.inspect() and DomainProcessor.inspect() also say they return data frames.

**Pull request merged**